TESTS/netsocket/udp/udpsocket_echotest.cpp

```diff
@@ -183,20 +147,37 @@ void UDPSOCKET_ECHOTEST_NONBLOCK()
                 printf("[Round#%02d - Sender] error, returned %d\n", s_idx, sent);
                 continue;
             }
-            if (!tx_sem.try_acquire_for(WAIT2RECV_TIMEOUT * 2)) { // RX might wait up to WAIT2RECV_TIMEOUT before recvfrom
-                continue;
+
+            int recvd;
+            for (int retry_recv = 0; retry_recv <= RETRIES; retry_recv++) {
+                recvd = sock->recvfrom(NULL, rx_buffer, pkt_s);
+                if (recvd == NSAPI_ERROR_WOULD_BLOCK) {
+                    if (tc_exec_time.read() >= time_allotted) {
+                        break;
+                    }
+                    signals.wait_all(SIGNAL_SIGIO_RX, SIGIO_TIMEOUT);
+                    --retry_recv;
+                    continue;
+                } else if (recvd < 0) {
+                    printf("sock.recvfrom returned %d\n", recvd);
+                    TEST_FAIL();
+                    break;
+                } else if (recvd == pkt_s) {
+                    break;
+                }
+            }
+
+            if (recvd == pkt_s) {
+                break;
             }
-            break;
+            drop_bad_packets(*sock, 0); // timeout equivalent to set_blocking(false)
```

Comment:
What is this `drop_bad_packets()`?

I don't see any need for it.. If we did not receive all packets, then what is there to receive?

Reply:
Removed